modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -389,3 +389,106 @@ func (suite *KeeperTestSuite) assertUpgradeError(actualError, expError error) {
+			malleate: func() {
+				suite.chainA.DeleteKey(host.ChannelUpgradeKey(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
+			},
+			expPass:        false,
+			channelPresent: true,
+		},
+		{
+			name: "channel does not exist",
+			malleate: func() {
+				suite.chainA.DeleteKey(host.ChannelKey(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID))
+			},
+			expPass:        false,
+			channelPresent: false,
+		},
+		{
+			name: "fails with nil upgrade error",
+			malleate: func() {
+				upgradeError = nil
+			},
+			expPass:        false,
+			channelPresent: true,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			channelKeeper := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper
+
+			path.EndpointA.ChannelConfig.Version = mock.Version + "-v2"
+
+			suite.Require().NoError(path.EndpointA.ChanUpgradeInit())
+
+			upgradeError = types.NewUpgradeError(1, types.ErrInvalidChannel)
+
+			tc.malleate()
+
+			err := channelKeeper.AbortHandshake(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, upgradeError)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+
+				channel, found := channelKeeper.GetChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().True(found, "channel should be found")
+
+				suite.Require().Equal(types.OPEN, channel.State, "channel state should be %s", types.OPEN.String())
+				suite.Require().Equal(types.NOTINFLUSH, channel.FlushStatus, "channel flush status should be %s", types.NOTINFLUSH.String())
+				_, found = channelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().False(found, "upgrade info should be deleted")
+
+				errorReceipt, found := channelKeeper.GetUpgradeErrorReceipt(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				if upgradeError != nil {
+					suite.Require().True(found, "error receipt should be found")
+					suite.Require().Equal(upgradeError.GetErrorReceipt(), errorReceipt, "error receipt does not match expected error receipt")
+				}
+
+				_, found = channelKeeper.GetCounterpartyLastPacketSequence(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().False(found, "counterparty last packet sequence should not be found")
+
+			} else {
+				suite.Require().Error(err)
+				channel, found := channelKeeper.GetChannel(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				if tc.channelPresent {
+					suite.Require().True(found, "channel should be found")
+					suite.Require().Equal(types.INITUPGRADE, channel.State, "channel state should not be restored to %s", types.INITUPGRADE.String())
+				}
```

Comment:
Should there be here an `else` block to check `suite.Require().False(found, "channel should not be found")` in case the test case has `channelPresent` set to false?